pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -321,6 +321,26 @@ func (pt *PipelineTask) validateMatrixCombinationsCount(ctx context.Context) (er
 	return errs
 }
 
+func (pt PipelineTask) validateEmbeddedOrType() (errs *apis.FieldError) {
+	// Reject cases where APIVersion and/or Kind are specified alongside an embedded Task.
+	// We determine if this is an embedded Task by checking of TaskSpec.TaskSpec.Steps has items.
+	if pt.TaskSpec != nil && len(pt.TaskSpec.TaskSpec.Steps) > 0 {
```

Comment:
why is checking if there are steps necessary? can we just check `if pt.TaskSpec.TaskSpec != nil`?

Reply:
`pt.TaskSpec.TaskSpec` is just an inline `TaskSpec`, not a pointer, so this was the best I could come up with.